addons/purchase/static/src/components/purchase_file_uploader/purchase_file_uploader.js

```diff
@@ -35,6 +37,18 @@ export class PurchaseFileUploader extends Component {
         return this.props.list.getResIds(true);
     }
 
+    onClick(ev) {
+        const selectedRecords = this.records.filter(record => record.selected);
+        const vendorSet = new Set(selectedRecords.map(record => record.data.partner_id.id));
+        if (vendorSet.size > 1) {
+            this.dialog.add(WarningDialog, {
+                title: "Validation Error",
+                message: "You can only upload a bill for a single vendor at a time."
```

Comment:
Just a bit a cleaning and don't forget to translate those strings.
Also, you can access the selection directly from the list
```javascript
import { _t } from "@web/core/l10n/translation";
```
```suggestion
        const selectedRecords = this.props.list.selection;
        const vendorSet = new Set(selectedRecords.map((record) => record.data.partner_id.id));
        if (vendorSet.size > 1) {
            this.dialog.add(WarningDialog, {
                title: _t("Validation Error"),
                message: _t("You can only upload a bill for a single vendor at a time."),
```